### PR TITLE
Goodbye sidebar! (The templates to make this work)

### PR DESCRIPTION
**Pull request by thisisdano**
This updates the templates to remove the sidebar, and adds styling hooks for dashboard elements.

**Paired with https://github.com/18F/cg-style/pull/255**
**Based off Icon/Panel work in https://github.com/18F/cg-dashboard/pull/892 and https://github.com/18F/cg-style/pull/250 — Merge them first.**

- - -

screen shot 2016-12-30 at 7 27 00 am

- - -

screen shot 2016-12-30 at 7 27 18 am

### CODE REVIEW

File: static_src/components/disclaimer.jsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ export default class Disclaimer extends React.Component {
       flagImg = <img alt={ flagAlt } src={ flag }></img>;
     }
     return (
-      <div className={ this.styler('usa-disclaimer') }>
+      <div className={ this.styler('usa-disclaimer', 'disclaimer-dashboard') }>
```
Comment (thisisdano) [inline static_src/components/disclaimer.jsx]: hook to set dashboard-specific width

File: static_src/components/header.jsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ export default class Header extends React.Component {
       </Action>
     </HeaderLink>;
     return (
-    <header className={ this.styler('header') }>
+    <header className={ this.styler('header', 'header-dashboard') }>
```
Comment (thisisdano) [inline static_src/components/header.jsx]: hook to set dashboard-specific width/styling

File: static_src/components/main_container.jsx [hunk 3]
```diff
@@ -43,11 +43,9 @@ export default class App extends React.Component {
...
   render() {
     let content;
-    let sidebar;
 
     if (this.state.isLoggedIn) {
       content = this.props.children;
-      sidebar = <Nav />;
     } else {
       content = <Login />;
     }
```
Comment (thisisdano) [inline static_src/components/main_container.jsx]: removed sidebar references

File: static_src/components/main_container.jsx [hunk 4]
```diff
@@ -57,11 +55,8 @@ export default class App extends React.Component {
       <div>
         <Disclaimer />
         <Header />
-        <div className={ this.styler('sidenav-parent', 'main_content', 'content-dashboard') }>
-          <nav className={ this.styler('sidenav') }>
-            { sidebar }
-          </nav>
-          <main className={ this.styler('sidenav-main', 'usa-content') }>
+        <div className={ this.styler('main_content', 'content-dashboard') }>
+          <main className={ this.styler('usa-content') }>
```
Comment (thisisdano) [inline static_src/components/main_container.jsx]: removed sidebar references

**Comment by adborden** [comment]
@thisisdano are the `-no_sidebar` classes temporary? When would we want to remove them? Might be good to leave a comment to remind someone.

**Comment by thisisdano** [comment]
@adborden I wonder what we should do about those styles. I'd originally named them `-dashboard`, but Marco preferred this more 'universal' name. However, the purpose of the special style is to provide a hook specific to the dashboard width, which is probably narrower (not not _necessarily_ narrower) than the width of the marketing home/docs pages.

We will probably want/need a dashboard width that's distinct from the other pages. How best to style? I'd appreciate your thoughts!

**Comment by adborden** [comment]
Idk, with the sidenav sticking around for the docs site, I feel like the sidenav should somehow fit in the grid, then there would be two styles of layout, one "full-width" single column (dashboard) and then the second would be the two-column side-nav + main content layout (docs). If that was the case, then I would pick names like `-full` (single-column) and `-secondary` + `-main` (two-column).

I'm not really sure what to do in the meantime, maybe use `-full` to signify it's the full width, but it's a bit confusing because in contrast when using a sidenav you're using a non-standard grid where the main column is full-width anyway.

**Comment by adborden** [comment]
Hmm, now that I say that, I would make the full-widht version the first-class citizen, and the two-column layout the non-standard, by making that one require the `-with_sidebar`. That way it feels awkward and implies that there is work to do to make the docs site fit with the new grid.

**Comment by msecret** [comment]
@adborden maybe we should make your suggestions tech debt stories?

**Pull request merged**